src/datastore/pg-notifier.ts

```diff
@@ -56,38 +54,28 @@ type PgNotificationCallback = (notification: PgNotification) => void;
  */
 export class PgNotifier {
   readonly pgChannelName: string = 'stacks-api-pg-notifier';
-  subscriber: Subscriber;
+  readonly sql: PgSqlClient;
+  listener?: postgres.ListenMeta;
 
-  static create(usageName: string) {
-    return new PgNotifier(
-      getPgClientConfig({ usageName: `${usageName}:notifier`, pgServer: PgServer.primary })
-    );
+  static async create(usageName: string) {
+    const sql = await connectPostgres({ usageName: usageName, pgServer: PgServer.primary });
+    return new PgNotifier(sql);
   }
 
-  constructor(clientConfig: ClientConfig) {
-    this.subscriber = createPostgresSubscriber(clientConfig, {
-      native: false,
-      paranoidChecking: 30000, // 30s
-      retryLimit: Infinity, // Keep trying until it works or the API shuts down
-      retryTimeout: Infinity,
-      retryInterval: attempt => {
-        const retryMs = 1000;
-        logger.info(`PgNotifier reconnection attempt ${attempt}, trying again in ${retryMs}ms`);
-        return retryMs;
-      },
-    });
+  constructor(sql: PgSqlClient) {
+    this.sql = sql;
   }
 
   public async connect(eventCallback: PgNotificationCallback) {
-    this.subscriber.notifications.on(this.pgChannelName, message =>
-      eventCallback(message.notification)
-    );
-    this.subscriber.events.on('connected', () =>
-      logger.info(`PgNotifier connected, listening on channel: ${this.pgChannelName}`)
-    );
-    this.subscriber.events.on('error', error => logError('PgNotifier fatal error', error));
-    await this.subscriber.connect();
-    await this.subscriber.listenTo(this.pgChannelName);
+    try {
+      this.listener = await this.sql.listen(
+        this.pgChannelName,
+        message => eventCallback(JSON.parse(message) as PgNotification),
+        () => logger.info(`PgNotifier connected, listening on channel: ${this.pgChannelName}`)
+      );
+    } catch (error) {
+      logError('PgNotifier fatal connection error', error);
```

Comment:
Should this re-throw the error here after logging (which I think should exit the program)?

Reply:
Great point